core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -333,15 +333,9 @@ public KeyStore getPrimaryKeyStore() throws Exception {
                                 .getFirstProperty(RegistryResources.SecurityManagement.SERVER_PRIMARY_KEYSTORE_TYPE));
                 String password = config
                         .getFirstProperty(RegistryResources.SecurityManagement.SERVER_PRIMARY_KEYSTORE_PASSWORD);
-                FileInputStream in = null;
-                try {
-                    in = new FileInputStream(file);
+                try (FileInputStream in = new FileInputStream(file)) {
                     store.load(in, password.toCharArray());
-                    primaryKeyStore = store;
-                } finally {
-                    if (in != null) {
-                        in.close();
-                    }
+                    primaryKeyStore = KeyStoreUtil.convertKeystoreToJKS(store, password.toCharArray());
```

Comment:
The keystore conversion should be done only if the store.load(in, password.toCharArray()); is success. WDYT ?